source/expectations/data_values/ut_compound_data_helper.pkb

```diff
@@ -785,6 +785,17 @@ create or replace package body ut_compound_data_helper is
     return l_diffs;
   end;
   
+  function get_json_object(a_json_t json) return json_element_t is
+    l_obj json_element_t;
+  begin
+     $if dbms_db_version.version >= 21 $then
+       l_obj := case when a_json_t is null then cast (null as json_element_t ) else json_element_t.parse(json_query(a_json_t, '$' returning clob)) end;
+     $else
+       l_obj := null;
```

Comment:
this is not covered by tests.

Reply:
It has to be....
Its a main piece of code called when initiating json ...
It will not be covered below 21
```constructor function ut_data_value_json(self in out nocopy ut_data_value_json, a_value json) return self as result is
    l_value json_element_t := ut_compound_data_helper.get_json_object(a_value);
  begin 
    init(l_value);
    return;
  end;
```

Are you sure this is coverage from 21c version?

The second statement will never get called below 21 as there is no json type there other then dummy one.

Reply:
On 21 adding:
```
  function get_json_object(a_json_t json) return json_element_t is
    l_obj json_element_t;
  begin
     $if dbms_db_version.version >= 21 $then
       l_obj := case when a_json_t is null then cast (null as json_element_t ) else json_element_t.parse(json_query(a_json_t, '$' returning clob)) end;
     $else
       l_obj := null;
     $end
     dbms_output.put_line('Been here');
	 return l_obj;
  end;
```
and calling
```
declare
   l_actual   json;
    l_expected_message varchar2(32767);
    l_actual_message   varchar2(32767);
  begin
    -- Arrange
    l_actual   := json('{"Aidan Gillen": {"array": ["Game of Thrones","The Wire"],"string": "some string","int": "2","otherint": 4, "aboolean": "true", "boolean": false,"object": {"foo": "bar"}},"Amy Ryan": ["In Treatment","The Wire"],"Annie Fitzgerald": ["True Blood","Big Love","The Sopranos","Oz"],"Anwan Glover": ["Treme","The Wire"],"Alexander Skarsg?rd": ["Generation Kill","True Blood"],"Alice Farmer": ["The Corner","Oz","The Wire"]}');

    --Act
    ut3_develop.ut.expect( l_actual ).not_to_have_count( 6 );
end;
/
```
I got output:
```
Been here
FAILURE
  Actual: json [ count = 6 ] was expected not to have [ count = 6 ]
  at "anonymous block", line 10
```